### PR TITLE
issue_6221_FC_enhancement_instance_params

### PR DESCRIPTION
**Pull request by shmuelko**
Issue https://stage.practitest.com/p/4508/issues/255917
Web frontend PR: https://github.com/PractiTest/firecracker-web-frontend/pull/43

Add code to support creation of instance via step parameters or bdd parameters

### CODE REVIEW

File: src/practitest_firecracker/api.clj [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    (if (nil? uri)
+      results
+      (let [{:keys [status body]} (method uri params)]
+        (if (> attempts 0)
+          (case status
+            504 (do
+                  ;; load balancer freaking out, lets try again
+                  (Thread/sleep (* timeout-between-attempts 1000))
+                  (recur results uri (dec attempts) params))
+            502 (do
```
Comment (deshilovPT) [inline src/practitest_firecracker/api.clj]: 502, 500, 503 can be handled with single branch, something like

```
(500 502 503) (do 
              (log/warnf "%s responded with %s - %s more attempts" uri status attempts)
                  (Thread/sleep (* timeout-between-attempts 1000))
                  (recur results uri (dec attempts) params))
```

File: src/practitest_firecracker/core.clj [hunk 2]
```diff
@@ -50,13 +50,13 @@
 
             "create-and-populate-testset"
             (do
+              (println "Start Running Firecracker, Version: " fc-version)
```
Comment (deshilovPT) [inline src/practitest_firecracker/core.clj]: Is there a reason to use `println` other than `log/info` or something like this ?

File: src/practitest_firecracker/query_dsl.cljc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+(defn read-query [s]
+  (let [query    (edn/read-string s)
+        compiler (fn compile-query [query]
+                   (if (list? query)
+                     (let [[op & args] query]
+                       {:op   (compile-query op)
+                        :args (vec (map compile-query args))})
+                     query))]
+    #?(:cljs (if (and (not (number? query)) (not (string? query)))
```
Comment (deshilovPT) [inline src/practitest_firecracker/query_dsl.cljc]: Am I missing something here ? Those two branches look the same to me. Only thing that's different is `type` function on CLJ side, is there some hidden meta type information used ?

```
(or (= java.lang.Long (type query))
      (= java.lang.Double (type query))
      (and (= java.lang.String (type query)))
```

should be

`(or (string? query) (number? query))`

But maybe I'm missing something here.

File: src/practitest_firecracker/api.clj (hunk 1 above)
Comment (shmuelko) [reply src/practitest_firecracker/api.clj]: thank you, changed

File: src/practitest_firecracker/core.clj (hunk 2 above)
Comment (shmuelko) [reply src/practitest_firecracker/core.clj]: no reason, will change to log/info

File: src/practitest_firecracker/query_dsl.cljc (hunk 3 above)
Comment (shmuelko) [reply src/practitest_firecracker/query_dsl.cljc]: you are correct, I will merge this code into one line and will use "de morgan" on it

**Review by deshilovPT — APPROVED** [review]

**Review by Itaischv — APPROVED** [review]

**Pull request merged**